Installing package after BSOD fails with "hexadecimal value 0x00, is an invalid character. Line 1, position 1" error

### Checklist

- [x] I confirm there are no unresolved issues reported on the [Chocolatey Status page](https://status.chocolatey.org).
- [x] I have verified this is the correct repository for opening this issue.
- [x] I have verified no other issues exist related to my problem.
- [x] I have verified this is not an issue for a specific package.
- [x] I have verified this issue is not security related.
- [x] I confirm I am using **official**, and **not** unofficial, or modified, Chocolatey products.

### What You Are Seeing?

(I ticked the "I have verified no other issues exist related to my problem" box above but I did find related issue #1328 , but was asked to create a new issue regardless.)

I am trying to install a package - specifically notepadplusplus - and it fails because it is unable to parse some file that it expects to be in XML format. 

If I run this command - `choco install notepadplusplus --trace` this is the relevant portion of chocolatey.log (which contains all output seen onscreen as well):

```
2026-03-31 22:25:40,731 25728:1 [DEBUG] - Resolving resource PackageMetadataResource for source https://community.chocolatey.org/api/v2/ - :ResolveResource:0 
2026-03-31 22:25:40,735 25728:1 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='notepadplusplus.install',Version='8.9.3') - :MoveNext:0 
2026-03-31 22:25:40,736 25728:1 [DEBUG] - Resolving resource DownloadResource for source https://community.chocolatey.org/api/v2/ - :ResolveResource:0 
2026-03-31 22:25:40,752 25728:1 [DEBUG] - Attempting to delete file "". - :DeleteFile:0 
2026-03-31 22:25:40,752 25728:1 [INFO ] - Downloading package from source 'https://community.chocolatey.org/api/v2/' - :Install:0 
2026-03-31 22:25:40,753 25728:1 [DEBUG] - Package download location 'https://community.chocolatey.org/api/v2/package/notepadplusplus.install/8.9.3' - :Install:0 
2026-03-31 22:25:40,758 25728:1 [DEBUG] - Skipping package hash validation as feature 'usePackageHashValidation' is not enabled. - :ValidatePackageHash:0 
2026-03-31 22:25:40,786 25728:1 [ERROR] - notepadplusplus.install not installed. An error occurred during installation:
 '.', hexadecimal value 0x00, is an invalid character. Line 1, position 1. - :Install:0 
2026-03-31 22:25:40,790 25728:1 [INFO ] - notepadplusplus.install package files install failed with exit code 1. Performing other installation steps. - :HandlePackageResult:0 
2026-03-31 22:25:40,791 25728:1 [DEBUG] - No package information as package is null. - :Get:0 
2026-03-31 22:25:40,797 25728:1 [DEBUG] - No package information to save as package is null. - :Save:0 
2026-03-31 22:25:40,798 25728:1 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers... - :Publish:0 
2026-03-31 22:25:40,798 25728:1 [ERROR] - The install of notepadplusplus.install was NOT successful. - :HandlePackageResult:0 
2026-03-31 22:25:40,799 25728:1 [ERROR] - notepadplusplus.install not installed. An error occurred during installation:
 '.', hexadecimal value 0x00, is an invalid character. Line 1, position 1. - :HandleFailedOperation:0 
2026-03-31 22:25:40,807 25728:1 [WARN ] - 
Chocolatey installed 0/0 packages. 
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log). - :ReportActionSummary:0 
2026-03-31 22:25:40,808 25728:1 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers... - :Publish:0 
2026-03-31 22:25:40,810 25728:1 [ERROR] - Value cannot be null or an empty string.
Parameter name: value - :Main:0 
2026-03-31 22:25:40,810 25728:1 [ERROR] - More Details: System.ArgumentException: Value cannot be null or an empty string.
Parameter name: value
   at NuGet.Versioning.NuGetVersion.Parse(String value)
   at chocolatey.infrastructure.results.PackageResult.get_Identity()
   at chocolatey.infrastructure.app.services.NugetService.<>c__DisplayClass23_9.<Install>b__30(PackageDependency d)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at chocolatey.infrastructure.app.services.NugetService.Install(ChocolateyConfiguration config, Action`2 continueAction, Action`2 beforeModifyAction)
   at chocolatey.infrastructure.app.services.ChocolateyPackageService.Install(ChocolateyConfiguration config)
   at chocolatey.infrastructure.app.runners.GenericRunner.Run(ChocolateyConfiguration config, Container container, Boolean isConsole, Action`1 parseArgs)
   at chocolatey.infrastructure.app.runners.ConsoleApplication.Run(String[] args, ChocolateyConfiguration config, Container container)
   at chocolatey.console.Program.Main(String[] args) - :Main:0 
2026-03-31 22:25:40,811 25728:1 [DEBUG] - Exiting with 1 - :Main:0 
```

### What is Expected?

Ideally the installation should succeed, though that might be unreasonable given the circumstances (see below)

What I would more realistically expect would be for the installation process to clear whatever broken file or cache it is encountering.

If it can't do that, it could suggest a command that would allow me to force clear the relevant package.

Failing that, it should at least catch the exception and report which file it is having trouble with.

### How Did You Get This To Happen?

This is not a reproducible problem. I was running `choco upgrade all` when my computer had a blue-screen-of-death and restarted. At the time it was working on the Notepad++ package. On retrying I found some files that seemed corrupted so I deleted them. If I remember correctly, this included a registry.bad file which only contained a long string of the null character, no text or XML. This was after trying to use the suggestion here - https://docs.chocolatey.org/en-us/troubleshooting/#a-corrupt-registry-file-exists - which did nothing.

### User Story

As a user, when Chocolatey fails to install a package due to a problem parsing a file and cannot recover, it should suggest a command to fix the issue, or at least show me the exact file that it is failing on.

### System Details

* Operating System: Windows 11
* Windows PowerShell version: 5.1.26100.7920
* Chocolatey CLI Version: 2.7.0
* Chocolatey Licensed Extension version: n/a
* Chocolatey License type: free
* Terminal/Emulator: all commands were run in a Powershell window running as Administrator


### Installed Packages

```bash
Chocolatey v2.7.0
7zip 26.0.0
7zip.install 26.0.0
anki 25.9.20250923
audacity 3.7.7
autohotkey.portable 2.0.22
blender 5.1.0
chocolatey 2.7.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-misc-helpers.extension 0.0.4
chocolatey-windowsupdate.extension 1.0.5
clion-ide 2026.1.0
dbeaver 26.0.1
docker-desktop 4.66.1
dosbox 0.74.3
DotNet4.6.1 4.6.1055.20170308
dotnet4.7.1 4.7.2558.20190226
dotnetfx 4.8.0.20220524
dropbox 244.4.6580
epicgameslauncher 1.3.149
Everything 1.4.11032
f.lux 4.134.0
f.lux.install 4.134.0
filezilla 3.69.6
flow-launcher 2.1.1
foxitreader 2025.3.0.35737
git 2.53.0.2
git.install 2.53.0.2
goggalaxy 2.0.77.22
guitar-pro 7.6.0.2089
hg 6.3.1
hxd 2.5.0
iperf3 3.20.0
itch 26.1.9.20241215
iTunes 12.13.10.3
jetbrains-rider 2025.3.4
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
latencymon 7.31.0
libreoffice-fresh 26.2.1
musicbee 3.6.9202
netfx-4.6.2 4.6.2.20210905
obsidian 1.11.7
obs-studio 32.1.0
obs-studio.install 32.1.0
openmpt 1.32.6
playnite 10.51.0
procexp 17.10.0
procmon 4.1.0
putty 0.83.0
putty.portable 0.83.0
pycharm 2025.3.3
python313 3.13.12
rufus 4.13.0
scrivener 3.1.5.1
sourcetree 3.4.30
steam 3.10.91.91241028
streamdeck 7.3.1.22604
syncthing 2.0.15
teracopy 3.17.0
thunderbird 148.0.1
todolist 9.1.12
unity-hub 3.14.4
vcredist140 14.50.35719
vcredist2008 9.0.30729.616104
vcredist2010 10.0.40219.32503
vcredist2015 14.0.24215.20170201
vcredist2017 14.16.27052
vivaldi 7.9.3970.45
vivaldi.portable 7.9.3970.45
vlc 3.0.23
vlc.install 3.0.23
winmerge 2.16.54.2
wiztree 4.28.0
xnviewmp 1.10.3
xnviewmp.install 1.10.3
```

### Output Log

```bash
Sorry, the log is 10MB large and I don't have time to check the log for any personal data that needs scrubbing. Beyond the output above, I can also include the following config data which was output when the command was run:


2026-03-31 22:25:39,462 25728:1 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review. - :ToString:0 
2026-03-31 22:25:39,464 25728:1 [DEBUG] - Configuration: CommandName='install'|
CacheLocation='C:\Users\Ben\AppData\Local\Temp\chocolatey'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://community.chocolatey.org/api/v2/'|SourceType='normal'|
IncludeConfiguredSources='False'|ShowOnlineHelp='False'|Debug='False'|
Verbose='False'|Trace='True'|Force='False'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='True'|
DisableCompatibilityChecks='False'|AcceptLicense='False'|
AllowUnofficialBuild='False'|Input='notepadplusplus'|
AllVersions='False'|SkipPackageInstallProvider='False'|
SkipHookScripts='False'|PackageNames='notepadplusplus'|
Prerelease='False'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
UseHttpCache='True'|CacheExpirationInMinutes='30'|
AllowDowngrade='False'|ForceDependencies='False'|PinPackage='False'|
IncludeHeaders='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.26200.0'|
Information.PlatformName='Windows 11'|
Information.ChocolateyVersion='2.7.0.0'|
Information.ChocolateyProductVersion='2.7.0'|
Information.FullName='choco, Version=2.7.0.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Ben'|Information.UserDomainName='WIN11-DESKTOP'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|
Information.IsLicensedAssemblyLoaded='False'|
Information.LicenseType='Foss'|Information.CurrentDirectory='D:\'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
Features.UsePackageHashValidation='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.IgnorePinned='False'|
ListCommand.OrderBy='Id'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
ListCommand.ExplicitPageSize='False'|
ListCommand.ExplicitSource='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
UpgradeCommand.IgnorePinned='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='Unknown'|
ApiKeyCommand.Command='Unknown'|PinCommand.Command='Unknown'|
LicenseCommand.Command='Unknown'|OutdatedCommand.IgnorePinned='False'|
ExportCommand.IncludeVersionNumbers='False'|Proxy.BypassOnLocal='True'|
TemplateCommand.Command='unknown'|CacheCommand.Command='Unknown'|
CacheCommand.RemoveExpiredItemsOnly='False'| - :FindCommand:0
```

### Additional Context

_No response_

### Acceptance Criteria

_No response_

### Related Issues

- Issue #1328

┆Issue is synchronized with this [Clickup task](https://app.clickup.com/t/868jb91n5) by [Unito](https://www.unito.io)

--- a/src/chocolatey.tests.integration/Scenario.cs
+++ b/src/chocolatey.tests.integration/Scenario.cs
@@ -192,16 +192,22 @@ namespace chocolatey.tests.integration
             NUnitSetup.MockLogger.Messages.Clear();
         }
 
-        public static void AddFiles(IEnumerable<Tuple<string, string>> files)
+        public static void AddFiles(IEnumerable<(string name, string content)> files)
         {
             foreach (var file in files)
             {
-                if (_fileSystem.FileExists(file.Item1))
-                {
-                    _fileSystem.DeleteFile(file.Item1);
-                }
-                _fileSystem.WriteFile(file.Item1, file.Item2);
+                AddFile(file.name, file.content);
             }
+        }
+
+        public static void AddFile(string name, string content)
+        {
+            if (_fileSystem.FileExists(name))
+            {
+                _fileSystem.DeleteFile(name);
+            }
+
+            _fileSystem.WriteFile(name, content);
         }
 
         public static void CreateDirectory(string directoryPath)

--- a/src/chocolatey.tests.integration/scenarios/InstallScenarios.cs
+++ b/src/chocolatey.tests.integration/scenarios/InstallScenarios.cs
@@ -4208,6 +4208,79 @@ namespace chocolatey.tests.integration.scenarios
             }
         }
 
+        public class When_installing_a_package_with_corrupt_local_package_already_on_disk : ScenariosBase
+        {
+            private PackageResult _packageResult;
+
+            public override void Context()
+            {
+                base.Context();
+
+                var corruptPackageFolder = Path.Combine(Scenario.GetPackageInstallPath(), Configuration.PackageNames);
+                Scenario.CreateDirectory(corruptPackageFolder);
+
+                // Add a corrupt nuspec (full of null bytes) as an "already installed, but corrupted" package.
+                // The nupkg is intentionally missing here, since NuGet doesn't recognise the package as being "installed"
+                // even if files are present at the target install location, so it will blindly attempt to install a package.
+                var corruptNuspec = Path.Combine(corruptPackageFolder, Configuration.PackageNames + ".nuspec");
+                Scenario.AddFile(corruptNuspec, new string((char)0, 1024));
+            }
+
+            public override void Because()
+            {
+                Results = Service.Install(Configuration);
+                _packageResult = Results.FirstOrDefault().Value;
+            }
+
+            [Fact]
+            public void Should_not_install_where_install_location_reports()
+            {
+                Assert.That(_packageResult.InstallLocation ?? string.Empty, Is.Not.EqualTo(string.Empty));
+                Assert.That(_packageResult.InstallLocation, Does.Not.Exist);
+            }
+
+            [Fact]
+            public void Should_move_the_package_to_lib_bad_directory()
+            {
+                var packageDir = Path.Combine(Scenario.GetTopLevel(), "lib-bad", Configuration.PackageNames);
+
+                Assert.That(packageDir, Does.Exist);
+            }
+
+
+            [Fact]
+            public void Should_contain_a_warning_message_that_it_did_not_install_successfully()
+            {
+                MockLogger.Messages.Should().ContainKey(LogLevel.Warn.ToStringSafe())
+                    .WhoseValue.Should().Contain(m => m.Contains("0/1"));
+            }
+
+
+            [Fact]
+            public void Should_not_have_a_successful_package_result()
+            {
+                _packageResult.Success.Should().BeFalse();
+            }
+
+            [Fact]
+            public void Should_not_have_inconclusive_package_result()
+            {
+                _packageResult.Inconclusive.Should().BeFalse();
+            }
+
+            [Fact]
+            public void Config_should_match_package_result_name()
+            {
+                _packageResult.Name.Should().Be(Configuration.PackageNames);
+            }
+
+            [Fact]
+            public void Should_have_an_empty_version_string()
+            {
+                _packageResult.Version.Should().Be(string.Empty);
+            }
+        }
+
         public class When_installing_a_package_with_non_normalized_version : ScenariosBase
         {
             private PackageResult _packageResult;

--- a/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
+++ b/src/chocolatey.tests.integration/scenarios/PackScenarios.cs
@@ -60,7 +60,7 @@ namespace chocolatey.tests.integration.scenarios
 
             protected void AddFile(string fileName, string fileContent)
             {
-                Scenario.AddFiles(new[] { new Tuple<string, string>(fileName, fileContent) });
+                Scenario.AddFile(fileName, fileContent);
             }
         }
 
@@ -136,7 +136,7 @@ namespace chocolatey.tests.integration.scenarios
             {
                 Configuration = Scenario.Pack();
                 Scenario.Reset(Configuration);
-                Scenario.AddFiles(new[] { new Tuple<string, string>("myPackage.nuspec", GetNuspecContent()) });
+                Scenario.AddFile("myPackage.nuspec", GetNuspecContent());
 
                 if (!string.IsNullOrEmpty(ExpectedSubDirectory))
                 {
@@ -600,7 +600,7 @@ namespace chocolatey.tests.integration.scenarios
                 Scenario.Reset(Configuration);
                 Configuration.Version = "0.1.0";
                 Configuration.PackCommand.Properties.Add("commitId", "1234abcd");
-                Scenario.AddFiles(new[] { new Tuple<string, string>("myPackage.nuspec", NuspecContentWithVariables) });
+                Scenario.AddFile("myPackage.nuspec", NuspecContentWithVariables);
             }
 
             public override void Because()

--- a/src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs
+++ b/src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs
@@ -1823,7 +1823,11 @@ ATTENTION: You must take manual action to remove {1} from
 
             if (!string.IsNullOrWhiteSpace(packageResult.InstallLocation) && _fileSystem.DirectoryExists(packageResult.InstallLocation))
             {
-                var normalizedVersion = new NuGetVersion(packageResult.Version).ToNormalizedStringChecked();
+                // If there's no version information (certain edge cases on package install errors), fallback to an unknown version with a timestamp
+                // so this can't fail out unexpectedly.
+                var normalizedVersion = string.IsNullOrEmpty(packageResult.Version)
+                    ? "unknown-{0}".FormatWith(DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss"))
+                    : new NuGetVersion(packageResult.Version).ToNormalizedStringChecked();
                 var failuresFolder = _fileSystem.CombinePaths(ApplicationParameters.PackageFailuresLocation, packageResult.Name, normalizedVersion);
 
                 if (_filesService.MovePackageUsingBackupStrategy(packageResult.InstallLocation, failuresFolder, restoreSource: false))
@@ -1835,7 +1839,7 @@ ATTENTION: You must take manual action to remove {1} from
 
         private void RestorePreviousPackageVersion(ChocolateyConfiguration config, PackageResult packageResult)
         {
-            if (packageResult.InstallLocation == null)
+            if (string.IsNullOrEmpty(packageResult.InstallLocation) || string.IsNullOrEmpty(packageResult.Version))
             {
                 return;
             }

--- a/src/chocolatey/infrastructure.app/services/NugetService.cs
+++ b/src/chocolatey/infrastructure.app/services/NugetService.cs
@@ -50,6 +50,7 @@ using NuGet.Protocol.Core.Types;
 using NuGet.Resolver;
 using NuGet.Versioning;
 using static chocolatey.StringResources;
+using System.Xml;
 
 namespace chocolatey.infrastructure.app.services
 {
@@ -954,7 +955,6 @@ Please see https://docs.chocolatey.org/en-us/troubleshooting for more
                         }
                     }
 
-
                     try
                     {
                         //TODO, do sanity check here.
@@ -975,13 +975,18 @@ Please see https://docs.chocolatey.org/en-us/troubleshooting for more
                                    _nugetLogger, CancellationToken.None).GetAwaiter().GetResult())
                         {
                             ValidatePackageHash(config, packageDependencyInfo, downloadResult);
-
-                            nugetProject.InstallPackageAsync(
-                                packageDependencyInfo,
-                                downloadResult,
-                                projectContext,
-                                CancellationToken.None).GetAwaiter().GetResult();
-
+                            try
+                            {
+                                nugetProject.InstallPackageAsync(
+                                    packageDependencyInfo,
+                                    downloadResult,
+                                    projectContext,
+                                    CancellationToken.None).GetAwaiter().GetResult();
+                            }
+                            catch (XmlException)
+                            {
+                                throw new ApplicationException("An unexpected error was encountered parsing a malformed nuspec file. This may occur if corrupt files are present in the package's install directory.");
+                            }
                         }
 
                         var installedPath = nugetProject.GetInstalledPath(packageDependencyInfo);
@@ -1042,7 +1047,16 @@ Please see https://docs.chocolatey.org/en-us/troubleshooting for more
 
                         var logMessage = "{0} not installed. An error occurred during installation:{1} {2}".FormatWith(packageDependencyInfo.Id, Environment.NewLine, message);
                         this.Log().Error(ChocolateyLoggers.Important, logMessage);
-                        var errorResult = packageResultsToReturn.GetOrAdd(packageDependencyInfo.Id, new PackageResult(packageDependencyInfo.Id, version.ToFullStringChecked(), null));
+
+                        // Set this install path if the folder does exist, to ensure otherwise unrecoverable scenarios still move corrupt files
+                        // to lib-bad if they happen to be present, when the continueAction is set up to do so.
+                        var packageInstallPath = _fileSystem.CombinePaths(nugetProject.Root, pathResolver.GetPackageDirectoryName(packageDependencyInfo));
+                        if (!_fileSystem.DirectoryExists(packageInstallPath))
+                        {
+                            packageInstallPath = null;
+                        }
+
+                        var errorResult = packageResultsToReturn.GetOrAdd(packageDependencyInfo.Id, new PackageResult(packageDependencyInfo.Id, version.ToFullStringChecked(), packageInstallPath));
                         errorResult.Messages.Add(new ResultMessage(ResultType.Error, logMessage));
                         if (errorResult.ExitCode == 0)
                         {
@@ -1801,11 +1815,18 @@ Please see https://docs.chocolatey.org/en-us/troubleshooting for more
                                 {
                                     ValidatePackageHash(config, packageDependencyInfo, downloadResult);
 
-                                    nugetProject.InstallPackageAsync(
-                                        packageDependencyInfo,
-                                        downloadResult,
-                                        projectContext,
-                                        CancellationToken.None).GetAwaiter().GetResult();
+                                    try
+                                    {
+                                        nugetProject.InstallPackageAsync(
+                                            packageDependencyInfo,
+                                            downloadResult,
+                                            projectContext,
+                                            CancellationToken.None).GetAwaiter().GetResult();
+                                    }
+                                    catch (XmlException)
+                                    {
+                                        throw new ApplicationException("An unexpected error was encountered parsing a malformed nuspec file. This may occur if corrupt files are present in the package's install directory.");
+                                    }
                                 }
 
                                 var installedPath = nugetProject.GetInstalledPath(packageDependencyInfo);

